scripts/apply_diff_and_rename.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+
+# Add other repo as a remote: `git remote add -f coreth git@github.com:ava-labs/coreth.git`.
+# Usage: ./scripts/apply_diff_and_rename.sh coreth/master (or subnet-evm/master)
+
+set -e;
+set -u;
+
+ROOT=$(git rev-parse --show-toplevel);
+cd "${ROOT}";
+
+BASE="${1}";
+
+git diff .."${BASE}" --binary | git apply --whitespace=nowarn
+
+if [[ "${BASE}" == coreth* ]]; then
+    echo "Replacing coreth with subnet-evm"
+    sed_command='s!github.com/ava-labs/coreth!github.com/ava-labs/subnet-evm!g'
+else
+    echo "Replacing subnet-evm with coreth"
+    sed_command='s!github.com/ava-labs/subnet-evm!github.com/ava-labs/coreth!g'
+fi
+
+# TODO: improve this command that finds all the "coreth" references and replaces them with "subnet-evm"
+sed_inplace=(sed -i) # Linux
+if [[ $(uname) == "Darwin" ]]; then
+    sed_inplace=(sed -i '')
+fi
+LANG=C find . -type f \! -name 'apply_diff_and_rename.sh' \! -path './.git/*' \! -path './contracts/node_modules/*' -exec "${sed_inplace[@]}" -e "${sed_command}" {} \;
+gofmt -w .
+go mod tidy
+
+# Restore contracts/.gitignore
+git checkout -- contracts/.gitignore 2>/dev/null || true  # Ignore if the file doesn't exist
```

Comment:
Probably should remove these lines for less special casing and merge https://github.com/ava-labs/coreth/pull/638 before getting started instead